Data/ByteString.hs

```diff
@@ -50,6 +50,8 @@ module Data.ByteString (
         singleton,              -- :: Word8   -> ByteString
         pack,                   -- :: [Word8] -> ByteString
         unpack,                 -- :: ByteString -> [Word8]
+        fromStrict,             -- :: ByteString -> Lazy.ByteString
+        toStrict,               -- :: Lazy.ByteString -> ByteString
```

Comment:
On second thought, in the context of the `Data.ByteString` module, these seem like better names to me:

```suggestion
        toLazy,                 -- :: ByteString -> Lazy.ByteString
        fromLazy,               -- :: Lazy.ByteString -> ByteString
```

`Data.ByteString.Lazy` should keep the old names though.

Thoughts?!

Reply:
I like the new names better.

Reply:
It feels a bit weird to have the very same function under two different names. Is there any prior art?

Imagine having both modules in scope:
```haskell
import qualified Data.ByteString as BS
import qualified Data.ByteString.Lazy as BL
```

If `fromStrict` is the same entity, just re-exported, GHC suggests only one option for a hole `_ :: BS.ByteString -> BL.ByteString` - I do not even have to search for this function in haddocks! But if we define `toLazy = fromStrict` in `Data.ByteString`, GHC would be obliged to suggest both options. Now this is really confusing for a user: he does not know that it is just synonyms, so needs to check haddocks for both modules and painstakingly compare semantics.

Reply:
However, that means that folks who just want to trim the import lists don't win.  They have to change their code, and if they want a simple migration path to the new  `bytestring` releases that make the imports optional, they've more work to do.

One might even provide both sets of names.  `fromStrict == toLazy` and `fromLazy == toStrict`, and re-export both forms from the `Internal` module. So you can have either name from either source.

Reply:
My reason for proposing these names was that I think that with `Data.ByteString.toStrict` and `Data.ByteString.fromStrict` it's not obvious which type we're respectively converting from or to. The other type could plausibly be `[Word8]` or `String`.


> However, that means that folks who just want to trim the import lists don't win. They have to change their code, and if they want a simple migration path to the new `bytestring` releases that make the imports optional, they've more work to do.

@vdukhovni Can you expand on the scenario that you're describing here?

I don't understand how anyone could profit from the new `{from,to}Strict` re-exports while retaining compatibility with older bytestring versions.